osmaxx/core/templatetags/navigation.py

```diff
@@ -1,19 +1,34 @@
-# Adopted from https://djangosnippets.org/snippets/2783/
+from urllib.parse import urlsplit, urlunsplit
+
+import logging
 from django import template
 from django.conf import settings
+from django.http import HttpRequest
+
+logger = logging.getLogger(__name__)
 
 register = template.Library()
 
 
 @register.filter
-def siteabsoluteurl(absoluteurl, request):
+def siteabsoluteurl(relative_or_absolute_url: str, request: HttpRequest) -> str:
```

Comment:
This is not an argument to win or loose, I don't care enough to waste time arguing about this. If it helps you understand this piece of code, I'll accept it as it stands ;-)